proposed/http-client/http-client.md

```diff
@@ -41,9 +41,8 @@ use Psr\Http\Message\ResponseInterface;
 interface ClientInterface
 {
     /**
-     * Sends a PSR-7 request and returns a PSR-7 response. No response will get a special treatment. Ie: 
-     *   - Redirects will not be followed
-     *   - No exceptions will be thrown on 4xx or 5xx responses 
+     * Sends a PSR-7 request and returns a PSR-7 response. Every technically correct HTTP respons 
+     * MUST be returned as is, even if it represents a HTTP error response or a redirect instruction.
```

Comment:
Even for the 100 continue ?

Reply:
Yes, I think so. By the same arguments as we use on 301 responses. 
What do you think?

Reply:
No, definitely not for 100 continue responses, it doesn't make any sense. I have mentioned this earlier.

Reply:
so we would make the exception for status 100 that MUST be handled, but keep 3xx as MUST NOT be handled? i would agree with that. the meta should explain why (and probably mention that plugins then can handle the 3xx to follow redirects)

Reply:
We just have to link to https://tools.ietf.org/html/rfc7231#section-6.2.